### PR TITLE
Include label in the span of loops

### LINKED ISSUE #27042: Bad span for labeled loops
Rustc's parser does not include a loop's label in its span.

So the span associated to the expression
```rust
'somelabel: loop {
    whatev();
}
```
will contain only
```rust
loop {
    whatev();
}
```

cc @nrc

### PR DESCRIPTION
**Pull request by marcusklaas**
This closes https://github.com/rust-lang/rust/issues/27042.

I'd love to know if there's a way to make a regression test for this!

### CODE REVIEW

**Comment by P1start** [comment]
If you put the label on a separate line from the loop’s body, you can test that an arbitrary error’s span starts on the line of the label:

```rust
fn main() {
    let _: i32 =
        'a: //~ ERROR mismatched types
        loop { break };
    let _: i32 =
        'b: //~ ERROR mismatched types
        while true { break };
    let _: i32 =
        'c: //~ ERROR mismatched types
        for _ in None { break };
}
```

(I haven’t actually tested this, but it should work.)

**Comment by marcusklaas** [comment]
Thanks @P1start, that worked beautifully. The test even helped me catch a case I forgot about (while let).

File: src/test/compile-fail/issue-27042.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Regression test for #27042.
```
Comment (nrc) [inline src/test/compile-fail/issue-27042.rs]: Could you say what is actually being tested here (the issue number is clear from the file name).

**Comment by nrc** [comment]
r=me modulo the test comment nit

**Comment by marcusklaas** [comment]
Updated!

**Pull request merged**